test/functional/api/server_requests_spec.lua

```diff
@@ -32,6 +32,20 @@ describe('server -> client', function()
     end)
   end)
 
+  describe('NULL-initialized string handling', function()
+    it('works', function()
+      local function on_setup()
+        eq({'', ''}, eval('rpcrequest('..cid..', "nstring")'))
+        stop()
+      end
+
+      local function on_request(method, args)
+        return {'', ''}
```

Comment:
Don't know enough lua. I would assume, we are only interested in the request response which should be a list of empty strings (see #3844). Perhaps @snoe can help here?